### PR TITLE
New widget icons and alignment added (#3195)

### PR DESCRIPTION
**Pull request by alexlindroos**
Here is the PR for the new widget tree. Coloring of the widget tree and layout explorer is waiting for the new brand colors.

Before:
Screenshot 2021-07-23 at 16 08 39

After:

Screenshot 2021-07-23 at 15 56 45

### CODE REVIEW

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart [hunk 1]
```diff
@@ -42,9 +42,9 @@ extension InspectorColorScheme on ColorScheme {
       isLight ? Colors.yellowAccent : const Color.fromARGB(255, 70, 73, 76);
 }
 
-const double iconPadding = 5.0;
+const double iconPadding = 4.0;
 const double chartLineStrokeWidth = 1.0;
-double get columnWidth => isDense() ? 12 : 16.0;
+const double columnWidth = 16.0;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/inspector_tree.dart:47]: why is this removed? We use dense mode for embedding the inspector in the IDEs.

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart [hunk 2]
```diff
@@ -218,7 +218,7 @@ class InspectorTreeNode {
           ticks: ticks,
           depth: depth,
           lineToParent:
-              !node.isProperty && index != 0 && node.parent.showLinesToChildren,
+          !node.isProperty && index != 0 && node.parent.showLinesToChildren,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/inspector_tree.dart:221]: I think formatting is off here. Maybe make sure you are on the latest flutter and then run `flutter/bin/dart format .`

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart [hunk 3]
```diff
@@ -701,4 +701,4 @@ mixin InspectorTreeFixedRowHeightController on InspectorTreeController {
     targetRect = targetRect.inflate(20.0);
     scrollToRect(targetRect);
   }
-}
+}
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/inspector_tree.dart:704]: add back new line at end of file - formatter should take care of that

File: packages/devtools_app/lib/src/ui/icons.dart [hunk 4]
```diff
@@ -285,6 +291,111 @@ class ThemedImageIcon extends StatelessWidget {
...
+  static String getAssetName(String widgetType) {
+    if (widgetType == null) {
+      return null;
+    }
+
+    return _iconMap[_stripBrackets(widgetType)];
+  }
+
+  static String _stripBrackets(String widgetType) {
+    final bracketIndex = widgetType.indexOf('<');
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/icons.dart:304]: can you add a comment describing what the expected input looks like?

File: packages/devtools_app/test/icons_test.dart [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      expect(asset, 'icons/inspector/widget_icons/animated.png');
+    });
+
+    test('Null asset if widget is not in the icon map', () {
+      const String widgetName = 'CustomWidget';
+      final asset = WidgetIcons.getAssetName(widgetName);
+      expect(asset, null);
+    });
+  });
+}
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/icons_test.dart:22]: new line at end of file

File: packages/devtools_app/test/icons_test.dart [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/icons_test.dart:1]: nit: 2021

File: packages/devtools_app/lib/src/ui/icons.dart [hunk 7]
```diff
@@ -53,24 +53,30 @@ class CustomIcon extends StatelessWidget {
 }
 
 class CustomIconMaker {
-  final Map<String, CustomIcon> iconCache = {};
+  final Map<String, Widget> iconCache = {};
 
-  CustomIcon getCustomIcon(String fromText,
+  Widget getCustomIcon(String fromText,
       {IconKind kind, bool isAbstract = false}) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/icons.dart:59]: not your fault, but trailing comma after `isAbstract = false`

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart (hunk 2 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/inspector/inspector_tree.dart]: fixed

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart (hunk 3 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/inspector/inspector_tree.dart]: fixed

File: packages/devtools_app/lib/src/ui/icons.dart (hunk 4 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/ui/icons.dart:309]: Yea I will

File: packages/devtools_app/test/icons_test.dart (hunk 5 above)
Comment (alexlindroos) [reply packages/devtools_app/test/icons_test.dart]: fixed

File: packages/devtools_app/test/icons_test.dart (hunk 6 above)
Comment (alexlindroos) [reply packages/devtools_app/test/icons_test.dart]: fixed

File: packages/devtools_app/lib/src/ui/icons.dart (hunk 7 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/ui/icons.dart]: fixed

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart (hunk 1 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/inspector/inspector_tree.dart]: sorry, put it back

**Comment by tomgilder** [comment]
@googlebot I consent

**Comment by tomgilder** [comment]
Sorry to push more changes to this PR, but I think it's better to review these changes all together.

The additional commit adds several more widget icons, and implements widget colors in the layout explorer. Because of showing the colors in layout explorer, the layout explorer now uses a thicker border and slight shadow to show the selected widget:

Screenshot_2021-07-29_at_23 59 40

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart [hunk 8]
```diff
@@ -0,0 +1,238 @@
+import 'package:flutter/material.dart';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart:1]: need copyright at top of file

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/utils.dart [hunk 9]
```diff
@@ -151,81 +155,94 @@ class WidgetVisualizer extends StatelessWidget {
...
-      ),
-      decoration: BoxDecoration(
-        border: Border.all(
-          color: borderColor,
         ),
-        color: theme.canvasColor.darken(),
-      ),
-    );
+      );
+    });
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/layout_explorer/ui/utils.dart:245]: nit: trailing comma

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+
+  static WidgetTheme fromName(String widgetType) {
+    if (widgetType == null) {
+      return const WidgetTheme();
+    }
+
+    return themeMap[_stripBrackets(widgetType)] ?? const WidgetTheme();
+  }
+
+  /// Strips the brackets of the widget
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart:20]: dartdoc nit. r/of/off and add new line after first sentence.

```
/// Strips the brackets off of the widget name.
///
/// For example: `AnimatedBuilder<String>` -> `AnimatedBuilder`.
```

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+    // Scroll
+    'CustomScrollView': scrollTheme,
+    'DraggableScrollableSheet': scrollTheme,
+    'SingleChildScrollView': scrollTheme,
+    'Scrollable': scrollTheme,
+    'Scrollbar': scrollTheme,
+    'ScrollConfiguration': scrollTheme,
+    'GridView': WidgetTheme(iconAsset: WidgetIcons.gridView),
+    'ListView': WidgetTheme(iconAsset: WidgetIcons.listView),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart:116]: put this WidgetTheme into a const above since it is used in multiple places

**Review by kenzieschmoll — APPROVED** [review]
LGTM after the last few comments are addressed

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart (hunk 11 above)
Comment (tomgilder) [reply packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart]: Done

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart (hunk 10 above)
Comment (tomgilder) [reply packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart]: Done

File: packages/devtools_app/lib/src/inspector/layout_explorer/ui/utils.dart (hunk 9 above)
Comment (tomgilder) [reply packages/devtools_app/lib/src/inspector/layout_explorer/ui/utils.dart]: Fixed

**Review by kenzieschmoll — APPROVED** [review]
